File: 187-repeated-dna-sequences/repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        seen = set()
        res = set()
        for i in range(len(s) - 9):
            if s[i:i+10] in seen:
                res.add(s[i:i+10])
            else:
                seen.add(s[i:i+10])
        return list(res)

File: 187-repeated-dna-sequences/repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        if (s.size() < 10)
            return {};
        std::unordered_set<std::string> seen;
        std::unordered_set<std::string> res;
        for (int i = 0; i < s.size() - 9; ++i) {
            std::string slice = s.substr(i, 10);
            if (seen.contains(slice) == false) {
                seen.insert(slice);
            }
            else {
                res.insert(slice);
            }
        }
        std::vector<std::string> result(res.begin(), res.end());
        return result;
    }
};